benchmarks/ekomark/benchmark/external/LHA_utils.py

```diff
@@ -83,18 +124,30 @@ def rotate_data(raw, rotate_to_evolution_basis=False):
             rotated data
     """
     inp = []
-    for l in raw_label_list:
+    i = "default"
+    if is_ffns_nnlo:
+        i = "NNLO_FFNS"
+    for l in raw_label_list[i]:
         inp.append(raw[l])
     inp = np.array(inp)
     if rotate_to_evolution_basis:
-        rot = np.dot(LHA_rotate_to_evolution, inp)
-        return dict(zip(evol_label_list, rot))
+        if is_ffns_nnlo:
+            rot = np.dot(LHA_rotate_to_evolution_NNLO_FFNS, inp)
+            return dict(zip(evol_label_list[2], rot))
+        else:
+            rot = np.dot(LHA_rotate_to_evolution, inp)
+            return dict(zip(evol_label_list[1], rot))
     else:
-        rot = np.dot(LHA_rotate_to_flavor, inp)
+        if is_ffns_nnlo:
+            rot = np.dot(LHA_rotate_to_flavor_NNLO_FFNS, inp)
+        else:
+            rot = np.dot(LHA_rotate_to_flavor, inp)
```

Comment:
Same of above: in principle you need 2 decoupled `if`, instead of 3 ones, with 2 nested into the first:
- before check if `is_ffns_nnlo` and define `rot` to rotate to flavor basis
- if `rotate_to_evolution_basis` update the rotation multiplying by the suitable matrix

If the default one should be flavor or evolution it's completely arbitrary, but since the variable is `rotate_to_evolution_basis` I took the flavor one as a more natural default.